### PR TITLE
Modify Order table

### PR DESCRIPTION
**Pull request by islean**
## Description

This is a revision on the order table in which we would like to group the orders in CiGRID based on the pipeline, hence why the pipeline column is now being added. I also added a relationship from order to customer to be able to access more customer data.

### Added

- Pipeline column to the order table.
- Order-Customer relationship

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b modify-order -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by islean** [comment]
Alembic revisions work locally.

File: cg/server/dto/orders/orders_response.py
Comment (seallard) [inline cg/server/dto/orders/orders_response.py:7]: I'd just call this field `customer_name`

Comment (islean) [reply cg/server/dto/orders/orders_response.py:7]: Reasonable, but might be confusing when we have a Customer.name field as well?

File: alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py
Comment (islean) [inline alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py:25]: Probably should not be nullable?

File: tests/store/crud/read/test_read.py
Comment (seallard) [inline tests/store/crud/read/test_read.py:1555]: Is this test really necessary? I think we end up testing the sqlalchemy library here rather than our logic

File: cg/server/dto/orders/orders_response.py
Comment (seallard) [inline cg/server/dto/orders/orders_response.py:7]: You probably want to add the pipeline here as well, right? 😄

**Review by seallard — APPROVED** [review]
Nice 👍

File: cg/server/dto/orders/orders_response.py
Comment (seallard) [reply cg/server/dto/orders/orders_response.py:7]: Ah, we do?! Maybe we should display that instead in the frontend? Your call!

File: cg/services/orders/utils.py
Comment (seallard) [inline cg/services/orders/utils.py:5]: Oof, I'd not name this `PydanticOrder` 😢  But stick to it if you want. If it is necessary to have a different name than `Order`, I'd add a `Dto` or something to it

File: cg/server/dto/orders/orders_response.py
Comment (henrikstranneheim) [reply cg/server/dto/orders/orders_response.py:7]: Whenever guarantied to be unique I think `_id` should be used.

**Review by henrikstranneheim — APPROVED** [review]
Nice, just want to discuss using workflow instead of pipeline.

**Comment by Vince-janv** [comment]
Quick question on Database normalisation. In this case not duplicating data. Do we think pipeline should be removed from case in the future?

File: alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py
Comment (islean) [reply alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py:24]: I will change it!

**Comment by islean** [comment]
> Quick question on Database normalisation. In this case not duplicating data. Do we think pipeline should be removed from case in the future?

I would love that, after all cases have been connected to an order.

File: cg/server/dto/orders/orders_response.py
Comment (islean) [reply cg/server/dto/orders/orders_response.py:9]: Yes! That would be ideal 😅

File: alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py
Comment (seallard) [reply alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py]: Maybe order type is a better name for it?

File: tests/store/crud/read/test_read.py
Comment (islean) [reply tests/store/crud/read/test_read.py:1555]: Probably more for my own dev purposes, like a log message which should be removed before merging 😅

Comment (islean) [reply tests/store/crud/read/test_read.py]: removed

File: cg/services/orders/utils.py
Comment (islean) [reply cg/services/orders/utils.py]: Changed it to Order, and we'll see if anyone complains in the future.

File: alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py
Comment (islean) [reply alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py]: Have we not started out with workflow in other places though?

Comment (islean) [reply alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py]: For example /cg/meta/workflow which contains pipelines and fastq etc

Comment (ChrOertlin) [reply alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py]: indeed workflow is best here

**Comment by islean** [comment]
Seems to work for the new CiGRID page purposes:

image

File: cg/server/dto/orders/orders_response.py
Comment (islean) [reply cg/server/dto/orders/orders_response.py:9]: I will keep it as the internal id for now - I think production more commonly refers to customers with custXXX than the name chosen by the customers themselves.

File: alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py
Comment (islean) [reply alembic/versions/2024_01_25_d241d8c493fb_add_pipeline_to_order.py]: I will go with workflow as OrderType is a slightly different field in the order forms.

**Comment by islean** [comment]
Deployed to production (Hasta and cg-prod-services):

```
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/isak.ohlsson
done.
Log deploy... done.
cg, version 56.0.1
θ69° [isak.ohlsson@hasta:~] [P_base] 7m31s $
```

**Pull request merged**
